### PR TITLE
Discourage kernel.h usage and other header cleanups

### PR DESCRIPTION
**Pull request by ashevche**
We do not want to encourage people to use garbage which is kernel.h. The rule of thumb is to include the headers we are the direct users of. Clean up a bit the examples and text  of the book.

**UPDATE**
Besides that the examples have rather random headers. This has to be revisited and updated. In this PR it's a subset of the fixes of the mess.

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Please run `clang-format -i examples/*.c` to enforce consistent indention.

**Review by linD026 — APPROVED** [review]
Looks good to me.

For the clang-format:
```
diff --git a/examples/hello-5.c b/examples/hello-5.c
index e2cb42c..3447855 100644
--- a/examples/hello-5.c
+++ b/examples/hello-5.c
@@ -2,7 +2,7 @@
  * hello-5.c - Demonstrates command line argument passing to a module.
  */
 #include <linux/init.h>
-#include <linux/kernel.h>      /* for ARRAY_SIZE() */
+#include <linux/kernel.h> /* for ARRAY_SIZE() */
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/printk.h>
diff --git a/examples/intrpt.c b/examples/intrpt.c
index c35e02e..beeddd6 100644
--- a/examples/intrpt.c
+++ b/examples/intrpt.c
@@ -10,7 +10,7 @@
 
 #include <linux/gpio.h>
 #include <linux/interrupt.h>
-#include <linux/kernel.h>      /* for ARRAY_SIZE() */
+#include <linux/kernel.h> /* for ARRAY_SIZE() */
 #include <linux/module.h>
 #include <linux/printk.h>
```

**Comment by ashevche** [comment]
@linD026 Thank you for review, I will update later. Maybe I will have time to check more examples.

**Comment by jserv** [comment]
> @linD026 Thank you for review, I will update later. Maybe I will have time to check more examples.

Please resolve the compilation errors reported by GitHub Actions.

File: examples/example_tasklet.c [hunk 1]
```diff
@@ -2,7 +2,6 @@
  * example_tasklet.c
  */
 #include <linux/delay.h>
-#include <linux/interrupt.h>
```
Comment (linD026) [inline examples/example_tasklet.c:5]: Do not remove this header since the example still uses the functions/macros, e.g., `tasklet_kill()`, in this header.

**Review by linD026 — CHANGES_REQUESTED** [review]

**Comment by linD026** [comment]
For the others, it looks good to me.

File: examples/example_tasklet.c (hunk 1 above)
Comment (linD026) [reply examples/example_tasklet.c:5]: And, I think this is why GitHub Actions is yelling.

Comment (ashevche) [reply examples/example_tasklet.c:5]: Indeed. That's why as I said the examples needs to be revisited :-)
Thank you for your reviews!

**Comment by jserv** [comment]
Thank @ashevche for contributing!

**Pull request merged**